impl/src/main/java/org/kuali/rice/kim/web/struts/action/IdentityManagementGroupDocumentAction.java

```diff
@@ -228,6 +229,14 @@ public ActionForward addMember(ActionMapping mapping, ActionForm form, HttpServl
         }
         if(checkKimDocumentGroupMember(newMember) && 
         		KRADServiceLocatorWeb.getKualiRuleService().applyRules(new AddGroupMemberEvent("", groupDocumentForm.getGroupDocument(), newMember))){
+
+			// Set member's fullname if possible
+			Entity entity = KimApiServiceLocator.getIdentityService().getEntityByPrincipalName(newMember.getMemberName());

```

Comment:
Question: is there a possibility that both if statements (this one and the immediately previous one) would be true? In that case we'd have two redundant calls to IdentityService.getPrincipalByPrincipalName which I recall are very costly. If this is the case, is there any way we can consolidate to just one call? (on none... even better)